lib/js/jsfetch.nim

```diff
@@ -0,0 +1,78 @@
+## - Fetch for the JavaScript target: https://developer.mozilla.org/docs/Web/API/Fetch_API
+from asyncjs import PromiseJs
+from httpcore import HttpMethod
+export PromiseJs, HttpMethod
+
+when not defined(js) and not defined(nimdoc):
+  {.fatal: "Module jsfetch is designed to be used with the JavaScript backend.".}
+when defined(nodejs):
+  {.warning: "By design 'fetch()' is defined on the web browser, but may not be on NodeJS.".}
+
+template method2cstring(metod: HttpMethod): cstring =
+  ## Template that takes an `HttpMethod` and returns an *Uppercase* `cstring`,
+  ## but *only* for the HTTP Methods that are supported by the fetch API.
+  ## High performance and minimal code compared to just `$(HttpMethod)`.
+  assert metod notin {HttpTrace, HttpOptions, HttpConnect}, "HTTP Method not supported by fetch API"
+  case metod
+  of HttpHead:   "HEAD".cstring
+  of HttpGet:    "GET".cstring
+  of HttpPost:   "POST".cstring
+  of HttpPut:    "PUT".cstring
+  of HttpDelete: "DELETE".cstring
+  of HttpPatch:  "PATCH".cstring
+  else:          "GET".cstring
+
+func fetch*(url: cstring): PromiseJs {.importcpp: "fetch(#)".}
+  ## `fetch()` API, Simple GET only (generates `fetch(url)`).
+  ##
+  ## .. code-block:: nim
+  ##   doAssert fetch("https://nim-lang.org") is PromiseJs
+
+func fetch*(url: cstring, httpMethod: static[HttpMethod], body: cstring): PromiseJs {.asmNoStackFrame.} =
+  ## `fetch()` API, for any HTTP Method (static overload for optimization).
+  ##
+  ## .. code-block:: nim
+  ##   doAssert fetch("https://my.api.org", HttpPost, "body") is PromiseJs
+  # See the generated code if you dont understand why this overload exists.
+  assert url.len > 0, "url must not be empty string"
```

Comment:
**generally applicable comment**
this comment `"url must not be empty string"` adds 0 value. If the assert fails, the expression `url.len > 0` already tells you that, more succintly. The only value of such comments are either to explain something non-trivial, or show runtime values that can't be deduced from looking at source code. But since url.len would be 0 in this case, it's pointless to show `url` value.

Reply:
I agree, but good luck translating `url.len>0` on google translate or similar tho,
while you can with `"url must not be empty string"`, it is for Debug mode anyways.

Reply:
even if you google translated it, it wouldn't help a non developper end user much as they'd probably have no idea what generated that empty url in the first place; most likely some application logic that led there. asserts are for application logic validation, not user input validation (that's what exceptions are for); and anyone interested in looking into why an assert failed will have exactly the same information with or without this msg "url must not be empty string"
The only useful assert msgs IMO are ones that show useful runtime contextual info.
But I won't insist further if you want to keep that msg.